files/lib/form/FormBuilder.class.php

```diff
@@ -284,24 +277,19 @@ protected function readParameter($needle, $haystack, $type = 'string')
     }
 
     /**
-     * Calls the 'saved' event after the successful call of the save method.
-     * This functions won't called automatically. You must do this manually, if you inherit AbstractForm.
-     *
-     * @see \wcf\form\AbstractForm::saved()
-     */
-    protected function saved() 
-    {
-        parent::saved();
-    }
-
-    /**
      * Reads/Gets the data to be displayed on this page.
      *
      * @see \wcf\page\IPage::readData()
      */
     public function readData() 
     {
         parent::readData();
+        //add values if this a edit form page
+        if($this->requiresValidObject && empty($_POST)){
+			foreach($this->buildAttributeList() as $name => $options){
+				$this->valueList[$name] = $this->object->$name;
+			}
+		}
```

Comment:
I removed this part. Values from the attributeList can be marked as `skip`, so that they don't end up being saved with the model. Users might have properties in their form, that don't even have a correspondence in the object, so we'd be trying to access properties that don't exist in line 290.

Values from the object (either on initial load or after a save) are available by the `$object` variable that's being passed into the template, e.g. `$object->name`.